lib/mumuki/classroom/models/submission.rb

```diff
@@ -18,20 +17,24 @@ class Submission
   field :manual_evaluation, type: String
   field :origin_ip, type: String
 
-  embeds_many :messages
+  embeds_many :messages, class_name: 'Mumuki::Classroom::Message'
+
+  embedded_in :assignment, class_name: 'Mumuki::Classroom::Assignment'
+  embedded_in :suggestion, class_name: 'Mumuki::Classroom::Suggestion'
+  embedded_in :last_assignment, class_name: 'Mumuki::Classroom::LastAssignment'
 
   def evaluate_manually!(comment, status)
     self.status = status
     self.manual_evaluation = comment
   end
 
   def add_message!(message)
-    self.messages << Message.new(message.as_json)
+    self.messages << Mumuki::Classroom::Message.new(message.as_json)
   end
 
   def expectation_results
     self[:expectation_results]&.map do |expectation|
-      {html: Mumukit::Inspection::I18n.t(expectation), result: expectation['result']}
+      {html: Mulang::Expectation.parse(expectation).translate, result: expectation['result']}
```

Comment:
The `translate` method takes optional parameters, that should be customized whenever possible. Please have a look at how laboratory deals with it

Reply:
Since we don't have access to domain exercise or language from here, this was not trivial. We've decided to just add a TODO for the future